common/io/TimeoutManager.cpp

```diff
@@ -94,11 +94,7 @@ void TimeoutManager::CancelTimeout(timeout_id id) {
 
 TimeInterval TimeoutManager::ExecuteTimeouts(TimeStamp *now) {
   Event *e;
-  if (m_events.empty())
-    return TimeInterval();
-
-  for (e = m_events.top(); !m_events.empty() && (e->NextTime() <= *now);
-       e = m_events.top()) {
+  while (!m_events.empty() && (e = m_events.top())->NextTime() <= *now) {
```

Comment:
A few more brackets for clarity:
```suggestion
  while (!m_events.empty() && ((e = m_events.top())->NextTime() <= *now)) {
```